indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -172,4 +194,232 @@ public static void setInputFormat(Job job, HadoopDruidIndexerConfig indexerConfi
       job.setInputFormatClass(TextInputFormat.class);
     }
   }
+
+  public static DataSegment serializeOutIndex(
+      final DataSegment segmentTemplate,
+      final Configuration configuration,
+      final Progressable progressable,
+      final TaskAttemptID taskAttemptID,
+      final File mergedBase,
+      final Path segmentBasePath
+  )
+      throws IOException
+  {
+    final FileSystem outputFS = FileSystem.get(segmentBasePath.toUri(), configuration);
+    final FileContext fileContext = FileContext.getFileContext(segmentBasePath.toUri(), configuration);
+    final Path tmpPath = new Path(segmentBasePath, String.format("index.zip.%d", taskAttemptID.getId()));
+    final AtomicLong size = new AtomicLong(0L);
+    final DataPusher zipPusher = (DataPusher) RetryProxy.create(
+        DataPusher.class, new DataPusher()
+        {
+          @Override
+          public void push() throws IOException
+          {
+            try (OutputStream outputStream = fileContext.create(
+                tmpPath,
+                EnumSet.of(CreateFlag.OVERWRITE, CreateFlag.CREATE),
+                Options.CreateOpts.createParent(),
+                Options.CreateOpts.bufferSize(256 * 1024)
+            )) {
+              size.set(zipAndCopyDir(mergedBase, outputStream, progressable));
+              outputStream.flush();
+            }
+            catch (IOException | RuntimeException exception) {
+              log.info(exception, "Exception in retry loop");
```

Comment:
I was torn on this one because if it is retrying that should perhaps just be a warning. But if something did go wrong here, I probably would want it to show up in ERROR, so I'll change it